blockchain/blocks/src/gossip_block.rs

```diff
@@ -1,19 +1,31 @@
 // Copyright 2019-2022 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
-use crate::BlockHeader;
+use crate::{ArbitraryCid, BlockHeader};
 use cid::Cid;
 use forest_encoding::tuple::*;
 use fvm_ipld_encoding::Cbor;
 
 /// Block message used as serialized `gossipsub` messages for blocks topic.
-#[derive(Clone, Debug, Serialize_tuple, Deserialize_tuple)]
+#[derive(Clone, Debug, PartialEq, Serialize_tuple, Deserialize_tuple)]
 pub struct GossipBlock {
     pub header: BlockHeader,
     pub bls_messages: Vec<Cid>,
     pub secpk_messages: Vec<Cid>,
 }
 
+impl quickcheck::Arbitrary for GossipBlock {
+    fn arbitrary(g: &mut quickcheck::Gen) -> Self {
+        let arbitrary_cids: Vec<ArbitraryCid> = Vec::arbitrary(g);
+        let cids: Vec<Cid> = arbitrary_cids.iter().map(|cid| cid.0).collect();
+        Self {
+            header: BlockHeader::arbitrary(g),
+            bls_messages: cids.clone(),
+            secpk_messages: cids,
```

Comment:
Using the same cids for both sets of messages leads to poor test coverage. For example, your code will never test having a different number of bls vs. secpk messages. And it'll never test having some messages of one kind but none of the other kind.

The goal with `Arbitrary` is to generate a structure with as close to a uniform sampling distribution as possible.

Reply:
OK, that makes sense. Fixing it up